virt-v2v/cold/entrypoint.go

```diff
@@ -0,0 +1,219 @@
+
+	//Disks on filesystem storage.
+	if err := LinkDisks(FS, 15); err != nil {
+		os.Exit(1)
+	}
+	//Disks on block storage.
+	if err := LinkDisks(Block, 10); err != nil {
+		os.Exit(1)
+	}
+
+	if source == vSphere {
+		if _, err := os.Stat("/etc/secret/cacert"); err == nil {
+			err = os.Symlink("/etc/secret/cacert", "/opt/ca-bundle.crt")
+			if err != nil {
+				fmt.Println("Error creating ca cert link ", err)
+				os.Exit(1)
+			}
+		} else {
+			err := os.Symlink("/etc/pki/tls/certs/ca-bundle.crt.bak", "/opt/ca-bundle.crt")
+			if err != nil {
+				fmt.Println("Error creating ca cert link ", err)
+				os.Exit(1)
+			}
+		}
+		virtV2vArgs = append(virtV2vArgs, "-ip /etc/secret/secretKey")
+
+		if info, err := os.Stat(VDDK); err == nil && info.IsDir() {
+			virtV2vArgs = append(virtV2vArgs,
+				"-it", "vddk",
+				"-io", fmt.Sprintf("vddk-libdir=%s", VDDK),
+				"-io", fmt.Sprintf("vddk-thumbprint=%s", os.Getenv("V2V_fingerprint")),
+			)
+		}
+		virtV2vArgs = append(virtV2vArgs, "--", os.Getenv("V2V_vmName"))
+	}
+
+	if err := executeVirtV2v(virtV2vArgs); err != nil {
+		fmt.Println("Error executing virt-v2v command ", err)
+		os.Exit(1)
+	}
+
+	if source == OVA {
+		var err error
+		xmlFilePath, err = getXMLFile(DIR, "xml")
+		if err != nil {
+			fmt.Println("Error gettin XML file:", err)
+			os.Exit(1)
+		}
+
+		http.HandleFunc("/xml", xmlHandler)
+		http.HandleFunc("/shutdown", shutdownHandler)
+		server = &http.Server{Addr: ":8080"}
+
+		fmt.Println("Starting server on :8080")
+		if err := server.ListenAndServe(); err != http.ErrServerClosed {
+			fmt.Printf("Error starting server: %v\n", err)
+			os.Exit(1)
+		}
+	}
+}
+
+func checkEnvVariablesSet(envVars ...string) bool {
+	for _, v := range envVars {
+		if os.Getenv(v) == "" {
+			return false
+		}
+	}
+	return true
+}
+
+func genName(diskNum int) string {
+	if diskNum <= 0 {
+		return ""
+	}
+
+	letters := []int32("abcdefghijklmnopqrstuvwxyz")
+	index := (diskNum - 1) % len(letters)
+	cycels := (diskNum - 1) % len(letters)
+
+	return genName(cycels) + string(letters[index])
+}
+
+func LinkDisks(diskKind string, num int) (err error) {
+	disks, err := filepath.Glob(diskKind)
+	if err != nil {
+		fmt.Println("Error getting disks ", err)
+		return
+	}
+
+	for _, disk := range disks {
+		diskNum, err := strconv.Atoi(disk[num:])
+		if err != nil {
+			fmt.Println("Error geting disks names ", err)
+			return err
+		}
+		diskLink := fmt.Sprintf("%s/%s-sd%s", DIR, os.Getenv("V2V_vmName"), genName(diskNum+1))
+		diskImgPath := disk
+		if diskKind == FS {
+			diskImgPath = fmt.Sprintf("%s/disk.img", disk)
+		}
+		if err = os.Symlink(diskImgPath, diskLink); err != nil {
+			fmt.Println("Error creating disk link ", err)
+			return err
+		}
+	}
+	return
+}
+
+func executeVirtV2v(args []string) (err error) {
+	virtV2vCmd := exec.Command("bash", "-c", strings.Join(args, " "), "|& /usr/local/bin/virt-v2v-monitor")
+	virtV2vCmd.Stdout = os.Stdout
+	virtV2vCmd.Stderr = os.Stderr
```

Comment:
just making sure - because of these two, we will see the whole output of virt-v2v in the container log, correct?

Reply:
Yes, that exactly that :)